### PR TITLE
gcp new filter access-approval

### PR DESCRIPTION
**Pull request by hiteshmck**

### CODE REVIEW

**Comment by kapilt** [comment]
analyzer ci failure is addressed in #8364

**Comment by hiteshmck** [comment]
> #8364

Ok, will wait till #8364 is merged and this current PR is reviewed.

File: tools/c7n_gcp/c7n_gcp/filters/__init__.py [hunk 1]
```diff
@@ -4,4 +4,5 @@
 from .metrics import GCPMetricsFilter
 from .sccfindings import SecurityComandCenterFindingsFilter
 from .iampolicy import IamPolicyFilter
-from .alerts import AlertsFilter
\ No newline at end of file
+from .alerts import AlertsFilter
+from .accessapproval import AccessApprovalFilter
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/filters/__init__.py:8]: resource specific filters go in the module that defined the resource, the filters package is typically for filters that can be used across multiple  resource types.

File: tools/c7n_gcp/c7n_gcp/filters/accessapproval.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.findings = self.get_findings()
+        return resources if self.match(self.findings) else []
+
+    def get_findings(self):
+        session = local_session(self.manager.session_factory)
+        client = session.client("accessapproval", "v1", "projects")
+        project = session.get_default_project()
+
+        try:
+            findings = client.execute_query(
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/filters/accessapproval.py:38]: findings is a dictionary but this is returning an empty array on error, an empty dictionary would at least match on type.

File: tools/c7n_gcp/c7n_gcp/filters/accessapproval.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                'getAccessApprovalSettings',
+                {'name': f"projects/{project}/accessApprovalSettings"},)
+        except HttpError as ex:
+            if (ex.status_code == 400
+                and ex.reason == "Precondition check failed.") \
+                    or (ex.status_code == 404):
+                """
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/filters/accessapproval.py:45]: inline typical style is to use a `#`  multi line strings are better suited as doc strings for modules, functions, and classes, not anonymous allocations inline.

File: tools/c7n_gcp/c7n_gcp/filters/accessapproval.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    def get_findings(self):
+        session = local_session(self.manager.session_factory)
+        client = session.client("accessapproval", "v1", "projects")
+        project = session.get_default_project()
+
+        try:
+            findings = client.execute_query(
+                'getAccessApprovalSettings',
+                {'name': f"projects/{project}/accessApprovalSettings"},)
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/filters/accessapproval.py:40]: this filter is being passed an array of projects, but its ignoring them to only check the default project for the execution, the individual projects should be checked.

File: tools/c7n_gcp/c7n_gcp/filters/accessapproval.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        filters:
+        - type: access-approval
+          key: enrolledServices.cloudProduct
+          value: "all"
+    """
+    schema = type_schema('access-approval', rinherit=ValueFilter.schema)
+    permissions = ('accessapproval.settings.get',)
+
+    def process(self, resources, event=None):
+        self.findings = self.get_findings()
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/filters/accessapproval.py:29]: not clear why this is being stored as an instance variable.

File: tools/c7n_gcp/c7n_gcp/filters/__init__.py (hunk 1 above)
Comment (hiteshmck) [reply tools/c7n_gcp/c7n_gcp/filters/__init__.py]: Thx, Moved it to appropriate gcp resource package.

File: tools/c7n_gcp/c7n_gcp/filters/accessapproval.py (hunk 2 above)
Comment (hiteshmck) [reply tools/c7n_gcp/c7n_gcp/filters/accessapproval.py]: Made it a dict, also changed the variable name to more appropriately suit this filter.

File: tools/c7n_gcp/c7n_gcp/filters/accessapproval.py (hunk 4 above)
Comment (hiteshmck) [reply tools/c7n_gcp/c7n_gcp/filters/accessapproval.py]: Thx for this valuable feedback, I refactored to check for individual projects.

File: tools/c7n_gcp/c7n_gcp/filters/accessapproval.py (hunk 5 above)
Comment (hiteshmck) [reply tools/c7n_gcp/c7n_gcp/filters/accessapproval.py]: Removed this redundant variable.

**Comment by hiteshmck** [comment]
@kapilt I incorporated PR feedback, please review when you get a chance.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**